src/pkg/egress/syslog/retryer.go

```diff
@@ -0,0 +1,52 @@
+package syslog
+
+import (
+	"log"
+	"time"
+
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
+)
+
+// RetryWriter wraps a WriteCloser and will retry writes if the first fails.
+type Retryer struct {
```

Comment:
Why not just represent `Retryer` as some new fields and methods in `HTTPSBatchWriter` since that's the only writer set to use it?